google-cloud-spanner/src/test/java/com/google/cloud/spanner/DatabaseTest.java

```diff
@@ -178,6 +198,72 @@ public void testBuildWithDatabaseDialect() {
     assertEquals(Dialect.GOOGLE_STANDARD_SQL, database.getDialect());
   }
 
+  @Test
+  public void testBuildWithProtoDescriptors() throws IOException {
+    final Database database =
+        dbClient
+            .newDatabaseBuilder(DatabaseId.of("my-project", "my-instance", "my-database"))
+            .setProtoDescriptors(PROTO_DESCRIPTORS_RESOURCE_PATH)
+            .build();
+
+    assertEquals(protoDescriptors, database.getProtoDescriptors());
+  }
+
+  @Test
+  public void testBuildWithProtoDescriptorsFromInputStream() throws IOException {
+    InputStream in =
+        getClass().getClassLoader().getResourceAsStream(PROTO_DESCRIPTORS_RESOURCE_PATH);
+    final Database database =
+        dbClient
+            .newDatabaseBuilder(DatabaseId.of("my-project", "my-instance", "my-database"))
+            .setProtoDescriptors(in)
+            .build();
+
+    assertEquals(protoDescriptors, database.getProtoDescriptors());
+  }
+
+  @Test
+  public void testBuildWithProtoDescriptorsFromByteArray() {
+    final Database database =
+        dbClient
+            .newDatabaseBuilder(DatabaseId.of("my-project", "my-instance", "my-database"))
+            .setProtoDescriptors(protoDescriptorsByteArray)
+            .build();
+
+    assertEquals(protoDescriptors, database.getProtoDescriptors());
+  }
+
+  @Test
+  public void testBuildWithProtoDescriptorsThrowsException() throws IOException {
+    InputStream in =
+        getClass().getClassLoader().getResourceAsStream(PROTO_DESCRIPTORS_RESOURCE_PATH);
+    in.close();
+    // case1: Test empty file path
+    assertThrows(
+        IllegalStateException.class,
+        () ->
+            dbClient
+                .newDatabaseBuilder(DatabaseId.of("my-project", "my-instance", "my-database"))
+                .setProtoDescriptors("")
+                .build());
+    // case2: Test invalid file path
```

Comment:
If the file path is invalid then we should more meaningful exception then `NullPointerException `

Reply:
Here `assertThrows` checks the class of exception that is returned. Thus, we check only for `NullPointerException`. 
However, the below is the actual exception that is thrown to user.
```
java.lang.NullPointerException: Input Proto Descriptors File path is invalid.
```